### PR TITLE
span: add ignored exceptions

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
## Description
Currently there is no way to ignore exceptions raised when a span is being used as a context manger however there are times when exceptions are used non-erroneously as in the case of [Http404 exceptions in Django](https://github.com/DataDog/dd-trace-py/pull/1834).

## Checklist
- [x] ~Entry added to release notes using [reno](https://docs.openstack.org/reno/latest/user/usage.html)~
- [x] Tests provided; and/or
- [x] ~Description of manual testing performed and explanation is included in the code and/or PR.~
- [x] ~Library documentation is updated.~
- [x] ~[Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).~

### CODE REVIEW

File: ddtrace/span.py [hunk 1]
```diff
@@ -372,6 +382,9 @@ def set_exc_info(self, exc_type, exc_val, exc_tb):
         if not (exc_type and exc_val and exc_tb):
             return  # nothing to do
 
+        if self._ignored_exceptions and exc_type in self._ignored_exceptions:
```
Comment (Kyle-Verhoog) [inline ddtrace/span.py:385]: I'm on the fence about whether we want to do a direct comparison (`exc_type in self._ignored_exceptions`) or do an `isinstance` check on each ignored exception.

File: ddtrace/span.py [hunk 2]
```diff
@@ -104,6 +106,14 @@ def __init__(
 
         self._context = context
         self._parent = None
+        self._ignored_exceptions = None  # type: Optional[List[Exception]]
+
+    def _ignore_exception(self, exc):
+        # (Exception) -> None
```
Comment (majorgreys) [inline ddtrace/span.py:112]: 👍  for the type annotations

File: ddtrace/span.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply ddtrace/span.py]: Decided with @majorgreys to go ahead with the isinstance check to handle cases when users might subclass the exceptions that we ignore

**Review by majorgreys — APPROVED** [review]
LGTM with the subclass check!

**Pull request merged**